### PR TITLE
Add comments and tests for recommender/forecasting classes

### PR DESCRIPTION
**Pull request by jcamachor**
# Pull Request

## Title

Add comments and tests for recommender/forecasting classes

---

## Description

**Issue link**: #61 

---

## Type of Change

_Mark an `x` in the boxes that apply:_

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [X] Refactor
- [X] Documentation update
- [X] Tests

---

## Testing

Existing and new tests.

### CODE REVIEW

File: recommender/forecasting/TimeSeriesForecaster.py [hunk 1]
```diff
@@ -16,7 +16,28 @@
 
 
 class TimeSeriesForecaster:
+    """
+    A class for forecasting time series data using various forecasting models.
+
+    Attributes:
+        window_splitter (float): Proportion of data to use for training.
```
Comment (ksaur) [inline recommender/forecasting/TimeSeriesForecaster.py:23]: @apavlen can you verify? i *think* this looks right?

Comment (apavlen) [reply recommender/forecasting/TimeSeriesForecaster.py:23]: this is correct

File: recommender/forecasting/TimeSeriesForecaster.py [hunk 2]
```diff
@@ -16,7 +16,28 @@
...
+        sp (int): Seasonal periodicity.
+        selected_forecaster (list): List of selected forecasters.
+        forecaster (object): The forecaster object.
+        forecaster_param_grid (dict): Parameter grid for the forecaster.
+        cv (SlidingWindowSplitter): Cross-validation splitter.
+        data_dir (str): Directory containing the data.
```
Comment (apavlen) [inline recommender/forecasting/TimeSeriesForecaster.py:29]: Directory containing training data.

File: recommender/forecasting/TimeSeriesForecaster.py [hunk 3]
```diff
@@ -16,7 +16,28 @@
...
+    """
+    A class for forecasting time series data using various forecasting models.
+
+    Attributes:
+        window_splitter (float): Proportion of data to use for training.
+        sp (int): Seasonal periodicity.
+        selected_forecaster (list): List of selected forecasters.
+        forecaster (object): The forecaster object.
+        forecaster_param_grid (dict): Parameter grid for the forecaster.
+        cv (SlidingWindowSplitter): Cross-validation splitter.
```
Comment (apavlen) [inline recommender/forecasting/TimeSeriesForecaster.py:28]: I don't see it as a parameter and I don't see it used after. I think we need to get rid of this maybe? (separate issue)

File: recommender/forecasting/TimeSeriesForecaster.py [hunk 4]
```diff
@@ -42,6 +69,16 @@ def set_forecaster(self, selected_forecaster):
         self.forecaster_param_grid = {"selected_forecaster": [selected_forecaster]}
 
     def get_prediction(self, data, number_of_points_to_predict):
+        """
+        Gets the prediction for the specified number of points.
+
+        Args:
+            data (pd.DataFrame): The input data for prediction.
+            number_of_points_to_predict (int): Number of future data points to predict.
```
Comment (apavlen) [inline recommender/forecasting/TimeSeriesForecaster.py:77]: We can leave as is or rename to forecasting horizon to be more consistent with time series forecasting terminology

File: recommender/forecasting/models/oracle.py [hunk 5]
```diff
@@ -7,11 +7,24 @@
 #
 import os
 import pandas as pd
-from datetime import datetime, timedelta
+from datetime import datetime
 
 
 class Oracle:
+    """
+    Oracle class for loading and processing performance data from CSV files.
```
Comment (apavlen) [inline recommender/forecasting/models/oracle.py:15]: Oracle class for loading, processing, and evaluating performance data from CSV files.

 This class is designed to assess recommender system policies by comparing them against an ideal theoretical predictor, allowing for the separation of errors attributable to prediction and the recommender itself.

**Review by apavlen — APPROVED** [review]
LGTM with some comments

**Pull request merged**